src/wp-includes/media.php

```diff
@@ -2077,9 +2079,12 @@ function wp_print_auto_sizes_contain_css_fix() {
 		return;
 	}
 
-	?>
-	<style>img:is([sizes="auto" i], [sizes^="auto," i]) { contain-intrinsic-size: 3000px 1500px }</style>
-	<?php
+	$handle = 'wp-img-auto-sizes-contain';
+	wp_register_style( $handle, false );
+	wp_add_inline_style( $handle, 'img:is([sizes="auto" i], [sizes^="auto," i]) { contain-intrinsic-size: 3000px 1500px }' );
+
+	// Make sure inline style is printed first.
+	array_unshift( wp_styles()->queue, $handle );
```

Comment:
This seems a bit hacky to me, but I understand the purpose. I assume there's not a better way to control the order of registered styles besides maybe making sure that the `wp_register_style()` call is made on an earlier hook, like `init`, correct?

Reply:
Exactly.